lib/internal/Magento/Framework/CompiledInterception/Generator/CompiledPluginList.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\CompiledInterception\Generator;
+
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Interception\PluginList\PluginList;
+use Magento\Framework\Interception\ObjectManager\ConfigInterface;
+use Magento\Framework\ObjectManager\Config\Reader\Dom;
+
+class CompiledPluginList extends PluginList
+{
+    /**
+     * CompiledPluginList constructor.
+     * @param $objectManager ObjectManager
+     * @param $scope
+     * @param null $reader
+     * @param null $omConfig
+     * @param null $cachePath
+     */
+    public function __construct(
+        $objectManager,
+        $scope,
+        $reader = null,
+        $omConfig = null,
+        $cachePath = null
+    ) {
+        if (!$reader || !$omConfig) {
+            $reader = $objectManager->get(Dom::class);
+            $omConfig = $objectManager->get(ConfigInterface::class);
+        }
+        parent::__construct(
+            $reader,
+            new StaticScope($scope),
+            new FileCache($cachePath),
+            new \Magento\Framework\ObjectManager\Relations\Runtime(),
+            $omConfig,
+            new \Magento\Framework\Interception\Definition\Runtime(),
+            $objectManager,
+            new \Magento\Framework\ObjectManager\Definition\Runtime(),
+            ['first' => 'global'],
+            'compiled_plugins_' . $scope,
+            new FileCache($cachePath)
+        );
```

Comment:
avoiding inheritance will make this class much nicer

Reply:
This class actually exists so we can use logic from `_loadScopedData` and `_inheritPlugins` methods of `PluginList` without modifying this class. It mostly removes few DI dependencies so interceptors can be compiled with as few dependencies as possible (because even core classes can have plugins in theory).

Alternatively I could just modify `PluginList` class but this way we keep original method intact.